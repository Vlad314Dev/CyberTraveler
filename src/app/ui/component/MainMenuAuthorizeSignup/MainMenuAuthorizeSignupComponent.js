import './MainMenuAuthorizeSignupStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import AuthorizeSignup from 'UIComponent/AuthorizeSignup';
import {
    AUTHORIZE_OPTION,
    BACK_OPTION,
    optionLabels
} from 'UIStore/MainMenu/MainMenuConfig';

class MainMenuAuthorizeSignupComponent extends PureComponent
{
    static propTypes = {
        setActiveOption: PropTypes.func.isRequired
    };

    setActiveOption = (optionName) => () => {
        const { setActiveOption } = this.props;
        setActiveOption(optionName)
    }

    render()
    {
        return (
            <>
                <AuthorizeSignup />
                <div block="MainMenuOption" elem="List">
                    {optionLabels.map((option, index) => {
                        if (option.key === BACK_OPTION) {
                            return <div 
                                block="MainMenuOption" 
                                elem="Wrapper" 
                                key={ index }
                            >
                                <span 
                                    block="MainMenuOption" 
                                    elem="Value"
                                    key={ index }
                                    onClick={ this.setActiveOption(AUTHORIZE_OPTION) }
                                >{ option.label }</span>
                            </div>
                        }
                        
                        return null;
                    })}
                </div>
            </>
        );
    }
}

export default MainMenuAuthorizeSignupComponent;
