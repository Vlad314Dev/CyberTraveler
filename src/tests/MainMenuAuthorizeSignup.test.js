/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import MainMenuAuthorizeSignupComponent from '../app/ui/component/MainMenuAuthorizeSignup/MainMenuAuthorizeSignupComponent';

describe('MainMenuAuthorizeSignup', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <MainMenuAuthorizeSignupComponent 
                setActiveOption={ jest.fn() }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <MainMenuAuthorizeSignupComponent 
                setActiveOption={ jest.fn() }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
