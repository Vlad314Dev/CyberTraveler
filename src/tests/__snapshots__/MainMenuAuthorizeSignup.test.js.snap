// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuAuthorizeSignup Should render layout 1`] = `
Array [
  <React.Fragment>
    <Memo(Connect(AuthorizeSignupContainer)) />
    <div
      className="MainMenuOption__List"
    >
      <div
        className="MainMenuOption__Wrapper"
      >
        <span
          className="MainMenuOption__Value"
          onClick={[Function]}
        >
          Back
        </span>
      </div>
    </div>
  </React.Fragment>,
]
`;
